src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,8 +29,33 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        String[] addressPartsList = address.split(", ");
+        this.block = new Block(addressPartsList[0]);
+        this.street = new Street(addressPartsList[1]);
+        this.unit = new Unit(addressPartsList[2]);
+        this.postalCode = new PostalCode(addressPartsList[3]);
     }
+    
+    public Block getBlock() {
```

Comment:
why would you return the objects here? if there is a need to edit the fields, you can use accessor methods, no?